### DISCUSSION TITLE
v0.2.4

### DISCUSSION THREAD

**Discussion by hjiangsu** [thunder-app/thunder#753] [Announcements]
Hey everyone, I know it's been a while since the last Thunder general release, but it is finally here! As always, there have been many improvements, additions, changes and fixes in this release which should make for a even better Thunder experience. If you have any suggestions or feedback, feel free to discuss it in the Thunder community or on GitHub.

A reminder for anyone who didn’t catch the previous announcements, Thunder is now officially available on Google Play and on the App Store! You should also be able to search up Thunder on the Play Store/App Store to find it. For more details about the launch, see the following posts:
- Google Play Announcement: https://lemmy.world/post/1631968
- App Store Announcement: https://lemmy.world/post/2926034

If you are using the Google Play version or App Store version, please note that it may take a bit of time before you receive the update. As always, this is an alpha release, so do expect bugs to pop up here and there. Now onto the update notes. This update brings a few major features:

- Complete overhaul of the Create Comment page - this redesign matches the new Create Post page introduced previously
  - Markdown actions and preview icons are now located at the bottom for better accessibility
  - Adds the ability to upload images to comments
- Post and comment drafts are now available - when you start a new post or comment, it will save the contents of the post/comment so that you can go back to it without losing your changes
- Instance status in account selection - you can now see the general status of your logged in accounts when you select your account in the profile switcher. This enables you to know if your instance is currently down
- Ability to access your saved comments - the account page has been changed slightly to allow viewing of saved contents posts/comments, as well as your own posts and comments 
- Improved management of blocked users and communities - you can now search for users and communities to block. You can access the blocked list by going to your account profile, and selecting the gear icon at the top left
- Added setting to reduce animations throughout the app - this feature is in its initial stages and feedback is appreciated here
- Added initial polish and spanish translations - if you'd like to contribute towards more translations and know a bit about .arb files, feel free to add them in!
- Ability to export and import your app settings - this feature is available in the General settings page

And much much more. I can't go through all the changes in the changelog, but for those who are curious, feel free to check out the full changelog below.

We have a Matrix space if you would like to join in on discussions: https://matrix.to/#/#thunderapp:matrix.org

## Added
* Show OP identification first before self/mod/admin by @micahmo in https://github.com/thunder-app/thunder/pull/569
* Show full text of a URL when activating tooltip on post in feed by @micahmo in https://github.com/thunder-app/thunder/pull/582
* Re-design comment creation screen by @coslu in https://github.com/thunder-app/thunder/pull/643
* Support image uploading when creating comments by @coslu in https://github.com/thunder-app/thunder/pull/643
* Show active user indicator when creating posts and comments by @coslu in https://github.com/thunder-app/thunder/pull/643
* Added bot identifier in comments header by @micahmo in https://github.com/thunder-app/thunder/pull/638
* Added access to saved comments in account page by @CTalvio in https://github.com/thunder-app/thunder/pull/519
* Added Polish translation by @Pazdikan in https://github.com/thunder-app/thunder/pull/659
* Added default user avatar when none is present by @coslu in https://github.com/thunder-app/thunder/pull/666
* Added lock indicator to posts and prevent commenting when post is locked  by @ajsosa in https://github.com/thunder-app/thunder/pull/676
* Added the ability to combine the post FAB with the comment navigation buttons by @micahmo in https://github.com/thunder-app/thunder/pull/648
* Show special user identifiers in post by @micahmo in https://github.com/thunder-app/thunder/pull/669
* Added importing/exporting of settings to JSON by @hjiangsu in https://github.com/thunder-app/thunder/pull/644
* Added instance status and latency indicators in account switcher by @micahmo in https://github.com/thunder-app/thunder/pull/655
* Added option to not dim read posts by @micahmo in https://github.com/thunder-app/thunder/pull/695
* Added Spanish translations by @ajsosa in https://github.com/thunder-app/thunder/pull/662
* Added Spanish translations by @ireneairyn in https://github.com/thunder-app/thunder/pull/674
* Ability to mark deleted replies as read by @micahmo in https://github.com/thunder-app/thunder/pull/693
* Disable or hide downvote actions when instance does not have downvotes enabled by @hjiangsu in https://github.com/thunder-app/thunder/pull/717
* Add drafts for posts and comments by @micahmo in https://github.com/thunder-app/thunder/pull/700
* Improvements to user/community block management page by @micahmo in https://github.com/thunder-app/thunder/pull/704
* Show sort icon in header/settings by @micahmo in https://github.com/thunder-app/thunder/pull/718
* Hide community name and show username in community view by @micahmo in https://github.com/thunder-app/thunder/pull/724
* Replying to comments will keep your position by @ajsosa in https://github.com/thunder-app/thunder/pull/725
* Add Thunder setting to reduce animations throughout the app by @hjiangsu in https://github.com/thunder-app/thunder/pull/730
* Highlight currently selected option in the sidebar by @micahmo in https://github.com/thunder-app/thunder/pull/731

## Changed
* Prioritize and identify default theme in Settings by @micahmo in https://github.com/thunder-app/thunder/pull/611
* Hide gesture customize hint when gestures are disabled in the Settings page by @micahmo in https://github.com/thunder-app/thunder/pull/599
* Text preview improvements in feed by @micahmo in https://github.com/thunder-app/thunder/pull/631
* Added markdown rendering to taglines and support for multiple taglines by @micahmo in https://github.com/thunder-app/thunder/pull/593
* Moved saved post star indicator before the title by @ajsosa in https://github.com/thunder-app/thunder/pull/684
* Comment indicators follow theme colors and accents by @tom-james-watson in https://github.com/thunder-app/thunder/pull/671
* Removed thumbnail progress indicators by @micahmo in https://github.com/thunder-app/thunder/pull/691
* Display sort name as subtitle and make sort button use generic sort icon on top bar by @tom-james-watson in https://github.com/thunder-app/thunder/pull/670
* Improved dismiss read functionality by @CTalvio in https://github.com/thunder-app/thunder/pull/698
* Do not grey out read own/saved posts in profile by @CTalvio in https://github.com/thunder-app/thunder/pull/645
* Improvements to error handling by @micahmo in https://github.com/thunder-app/thunder/pull/647
* Bring comments in profile/replies up to feature parity with posts by @CTalvio in https://github.com/thunder-app/thunder/pull/608
* Change instance/username entry keyboard type by @micahmo in https://github.com/thunder-app/thunder/pull/687
* Show pencil icon for edited posts in feed by @micahmo in https://github.com/thunder-app/thunder/pull/692
* Improved ability to refresh posts by @micahmo in https://github.com/thunder-app/thunder/pull/685
* Remove hero animations by @micahmo in https://github.com/thunder-app/thunder/pull/689
* Improved bottom pickers (sort by) by @micahmo in https://github.com/thunder-app/thunder/pull/686
* Adjust FAB sort icon to use default icon by @micahmo in https://github.com/thunder-app/thunder/pull/711
* Add semantic label to post combined FAB by @micahmo in https://github.com/thunder-app/thunder/pull/713
* Improve text/labels on profile page by @micahmo in https://github.com/thunder-app/thunder/pull/715
* Improve accent color picking in settings by @micahmo in https://github.com/thunder-app/thunder/pull/708
* Improve user colors in post comment headers by @micahmo in https://github.com/thunder-app/thunder/pull/710
* Improve pin/favorite/locked icons by @CTalvio in https://github.com/thunder-app/thunder/pull/719

## Fixed
* Handled issue with loading comments without a direct parent by @micahmo in https://github.com/thunder-app/thunder/pull/624
* Do not capture FAB summon gesture while disabled by @micahmo in https://github.com/thunder-app/thunder/pull/653
* Ensure snackbar appears over image viewer by @micahmo in https://github.com/thunder-app/thunder/pull/661
* Fixed infinite spinner for load more replies + toast by @ajsosa in https://github.com/thunder-app/thunder/pull/667
* Fixed NPE that broke commenting by @ajsosa in https://github.com/thunder-app/thunder/pull/681
* Fixed top post missing and dismiss read top post disappearing by @CTalvio in https://github.com/thunder-app/thunder/pull/697
* Fixed another comment spinner by @micahmo in https://github.com/thunder-app/thunder/pull/672
* Fixed some issues with the app bar by @micahmo in https://github.com/thunder-app/thunder/pull/682
* Fixed semantic labeling of sort picker by @micahmo in https://github.com/thunder-app/thunder/pull/654
* Limit set state on isOverridingSwipeGestureAction in comment_card (perf improvement) by @ajsosa in https://github.com/thunder-app/thunder/pull/694
* Added dimension information to thumbnails to reduce layout jank when using full screen images option by @hjiangsu in https://github.com/thunder-app/thunder/pull/712
* Set system back button to go to history on profile by @micahmo in https://github.com/thunder-app/thunder/pull/714
* Fix double tap zoom bug by @CTalvio in https://github.com/thunder-app/thunder/pull/720
* Restore the share/downloading indicator by @micahmo in https://github.com/thunder-app/thunder/pull/727
* Show all releases via in-app update by @micahmo in https://github.com/thunder-app/thunder/pull/733
* Fix issue where dismiss read FAB action would not work consistently by @hjiangsu in https://github.com/thunder-app/thunder/pull/743

## New Contributors
* @Pazdikan made their first contribution in https://github.com/thunder-app/thunder/pull/659
* @ireneairyn made their first contribution in https://github.com/thunder-app/thunder/pull/674

**Full Changelog**: https://github.com/thunder-app/thunder/compare/v0.2.3+16...0.2.4

<hr /><em>This discussion was created from the release <a href='https://github.com/thunder-app/thunder/releases/tag/0.2.4'>v0.2.4</a>.</em>

**Comment by Fmstrat** [comment]
@hjiangsu Release notes above say it's an alpha ;)

(Unless all releases are considered as such)

**Reply by micahmo** [reply]
@Fmstrat Yes we are still considering all releases alpha for now. 😊 (Not to be confused with the prereleases, which are the ones appended with a hyphen, like 0.2.4-1).

**Reply by Fmstrat** [reply]
Sounds good. Interesting.. I lost my contributor title. Odd.

**Reply by micahmo** [reply]
Hmm, that is strange. I know some things changed after the repo was moved to an org. Have you had a PR merged since then? You are still in the [contributors list](https://github.com/thunder-app/thunder/graphs/contributors).

@hjiangsu Could consider adding you to the org if you want (up to him of course).

**Reply by hjiangsu** [reply]
> Release notes above say it's an alpha 

Yup, like @micahmo mentioned, I still consider this to be in the alpha stages. The main reason for this is that nothing is really stable yet and there's a whole bunch of refactoring that still needs work/missing features (including potential layout and UI changes)

Hmm, interesting - your title shows up on the mobile app, but not on the desktop version. I can add you into the organization if you'd like! 

![image](https://github.com/thunder-app/thunder/assets/30667958/faa15ad6-036e-40a9-a9df-25d7e1bce766)

**Reply by Fmstrat** [reply]
Sure, why not, I plan on continuing to contribute. I have a PR in now for the next release, hah.
